client/pool.c

```diff
@@ -65,7 +65,7 @@ struct miner_pool_data {
 	struct nonce_hash *nonces;
 	xdag_hash_t last_min_hash;
 	long double mean_log_difficulty;
-	uint16_t bounded_task_counter;
+	uint32_t bounded_task_counter;
```

Comment:
CPUs work better with 32-bit integers. Better to use them if your purpose is not save the memory.